### ISSUE TITLE
Autodismissing notifications

### ISSUE THREAD

**Issue by adamwolf** [beeminder/beedroid-issues#257]
Background:

Android notifications were changed in v5 in a bunch of ways. There are Emergency Day Reminders, Lead Day Reminders, "Now Safe" Notifications, Timers, and Announcements.

In v5, when you have a beemergency day for a goal with Android notifications turned on, it generates an "Emergency Day Reminder" notification.  This notification has a countdown timer to the goal deadline (and that timer is actually driven by Android--we don't get to set very many specifics about it).  By default, if you dismiss the Beemergency, the notification is replaced by a Now Safe notification.

Using the Android notification settings, you can disable each type of notification.  This means if you don't like Now Safe notifications, for instance, you can turn them off in your Android settings.

If you dismiss the Beemergency and have Now Safe notifications turned off, the Emergency Day Reminder just sits there, being wrong, and the chronometer starts going backwards (!!!) after the goal deadline.

I think there is lots of opportunity for improvement here.

It seems like it makes sense to stop/remove the chronometer at the goal deadline for everyone.

At least one person has asked that Emergency Day Reminders dismiss themselves entirely when it is no longer an Emergency Day (and it's important enough to them that they've turned off Android notifications completely!). Having Emergency Day Reminders go away on their own seems reasonable and fine--but I know we've had folks say they never want to have the notifications dismiss themselves.

I have thought about it a bit, and I'm not sure I can rig up a way to do this without a setting somewhere.  @dreeves, what do you think?

See also:
- https://forum.beeminder.com/t/autodismissing-android-notifications/10947/7

**Comment by dreeves** [comment]
I don't think we can afford to dive in too deep right now but does this capture the theoretical choices we could make:

0. Status quo: no auto-dismiss, things are broken if you turn off Now Safe
1. No auto-dismiss, no ability to turn off Now Safe
2. Always auto-dismiss, no such thing as Now Safe
3. A setting for whether to auto-dismiss

And option 1 (which might theoretically be my first choice as the maximally paternalistic and anti-magic option) is maybe not possible due to Androidness? Which means option 3 is probably a can of worms. Which makes option 2 sure look appealing and maybe a big simplification of the code?

**Comment by adamwolf** [comment]
I don't think things are broken if you turn off Now Safe?

I think the quickest and easiest version of this is "autodismiss countdown timers" and let the user turn off/on Now Safe in Android Settings.  We can also disable Now Safe by default for new installs.

**Comment by dreeves** [comment]
👍 to disabling Now Safe by default (if we don't want to kill it altogether).
